lib/universal-search.js

```diff
@@ -23,9 +25,46 @@ Search.prototype = {
     WindowWatcher.stop();
   },
 
-  loadIntoWindow: function(win) {},
+  loadIntoWindow: function(win) {
+    console.log('loadIntoWindow start');
+    const document = win.document;
 
-  unloadFromWindow: function(win) {},
+    // Sets the app global per window.
+    if (win.universalSearch === undefined) {
+      Object.defineProperty(win, 'universalSearch', {
+        configurable: true,
+        value: {}
+      });
```

Comment:
I think it's actually good enough to do `win.universalSearch = win.universalSearch || {}`. I was looking at this recently and I don't have notes on what motivated this approach in the old popup--maybe some long-forgotten Gecko copypasta.